pkg/asset/manifests/infrastructure.go

```diff
@@ -82,13 +66,50 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 		},
 		Status: configv1.InfrastructureStatus{
 			InfrastructureName:   clusterID.InfraID,
-			Platform:             platform,
 			APIServerURL:         getAPIServerURL(installConfig.Config),
 			APIServerInternalURL: getInternalAPIServerURL(installConfig.Config),
 			EtcdDiscoveryDomain:  getEtcdDiscoveryDomain(installConfig.Config),
+			PlatformStatus:       &configv1.PlatformStatus{},
 		},
 	}
 
+	switch installConfig.Config.Platform.Name() {
+	case aws.Name:
+		config.Status.Platform = configv1.AWSPlatformType
+		config.Status.PlatformStatus.Type = configv1.AWSPlatformType
+		config.Status.PlatformStatus.AWS = &configv1.AWSPlatformStatus{
+			Region: installConfig.Config.Platform.AWS.Region,
+		}
+	case azure.Name:
+		config.Status.Platform = configv1.AzurePlatformType
+		config.Status.PlatformStatus.Type = configv1.AzurePlatformType
+		config.Status.PlatformStatus.Azure = &configv1.AzurePlatformStatus{
+			ResourceGroupName: fmt.Sprintf("%s-rg", clusterID.InfraID),
+		}
+	case gcp.Name:
+		config.Status.Platform = configv1.GCPPlatformType
+		config.Status.PlatformStatus.Type = configv1.GCPPlatformType
+		config.Status.PlatformStatus.GCP = &configv1.GCPPlatformStatus{
+			ProjectID: installConfig.Config.Platform.GCP.ProjectID,
+			Region:    installConfig.Config.Platform.GCP.Region,
+		}
+	case libvirt.Name:
+		config.Status.Platform = configv1.LibvirtPlatformType
+		config.Status.PlatformStatus.Type = configv1.LibvirtPlatformType
+	case none.Name:
+		config.Status.Platform = configv1.NonePlatformType
+		config.Status.PlatformStatus.Type = configv1.NonePlatformType
+	case openstack.Name:
+		config.Status.Platform = configv1.OpenStackPlatformType
+		config.Status.PlatformStatus.Type = configv1.OpenStackPlatformType
+	case vsphere.Name:
+		config.Status.Platform = configv1.VSpherePlatformType
+		config.Status.PlatformStatus.Type = configv1.VSpherePlatformType
+	default:
+		config.Status.PlatformStatus.Type = configv1.NonePlatformType
+	}
+	config.Status.Platform = config.Status.PlatformStatus.Type
```

Comment:
It seems pointless to have this assignment and the same assignment in the case statements. The assignment in the case statements will be undone by this. Which one we keep depends on whether they are always equal or if there can be skew depending on the platform.

Reply:
Good catch, agree.

Reply:
opened https://github.com/openshift/installer/pull/1934 to fix it.